route_linux.go

```diff
@@ -709,12 +712,13 @@ func RouteDel(route *Route) error {
 // Equivalent to: `ip route del $route`
 func (h *Handle) RouteDel(route *Route) error {
 	req := h.newNetlinkRequest(unix.RTM_DELROUTE, unix.NLM_F_ACK)
-	return h.routeHandle(route, req, nl.NewRtDelMsg())
+	_, err := h.routeHandle(route, req, nl.NewRtDelMsg())
+	return err
 }
 
-func (h *Handle) routeHandle(route *Route, req *nl.NetlinkRequest, msg *nl.RtMsg) error {
-	if (route.Dst == nil || route.Dst.IP == nil) && route.Src == nil && route.Gw == nil && route.MPLSDst == nil {
-		return fmt.Errorf("one of Dst.IP, Src, or Gw must not be nil")
+func (h *Handle) routeHandle(route *Route, req *nl.NetlinkRequest, msg *nl.RtMsg) ([][]byte, error) {
+	if req.NlMsghdr.Type != unix.RTM_GETROUTE && (route.Dst == nil || route.Dst.IP == nil) && route.Src == nil && route.Gw == nil && route.MPLSDst == nil {
+		return nil, fmt.Errorf("one of Dst.IP, Src, or Gw must not be nil")
```

Comment:
error message is not in sync with if condition. It's a AND of many conditions. So not "one of "

Reply:
I've updated the message to include a note about msgtype. I read the statement as checking that at least one of the options is set. How would you want the error message updated to reflect the if statement condition?

Reply:
You are right.
Can we have it as `"Either Dst.IP, Src.IP or Gw must be set"` ?

Reply:
sounds good to me. Updated.